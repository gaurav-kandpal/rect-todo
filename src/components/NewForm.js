import React ,{ useState }from 'react';

const NewForm = ({ addTodo }) => {
	const [title,setTitle] = useState('');
	const handleSubmit = (e) => {
		e.preventDefault();
		addTodo(title);
		setTitle('');
	}
	return(
	<div id="myDIV" data-testid="todoform" className="header">
		<h2>My To Do List</h2>
		<form  onSubmit={handleSubmit}>			
			<input type="text" value={title} required onChange ={(e) => setTitle(e.target.value)}/>
			<input type="submit" value="Add ToDo" />
		</form>
	</div>	
	);
}
export default NewForm;
