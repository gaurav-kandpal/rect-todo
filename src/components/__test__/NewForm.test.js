import React from 'react';
import ReactDom from 'react-dom';
import NewForm from './../NewForm';
import {render,cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from "react-test-renderer";
afterEach(cleanup);
it('Render without Crashing',() => {
	const div =  document.createElement("div");
	ReactDom.render(<NewForm></NewForm>,div)  
})
it('Render Correctly',() => {
	const{getByTestId} = render(<NewForm></NewForm>)
	expect(getByTestId('todoform').textContent).toBe('My To Do List');
})

it("matched snapshot",() => {
	const tree = renderer.create(<NewForm></NewForm>).toJSON();
	expect(tree).toMatchSnapshot();
})

