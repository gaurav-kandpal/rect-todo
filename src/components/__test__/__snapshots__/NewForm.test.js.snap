// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<div
  className="header"
  data-testid="todoform"
  id="myDIV"
>
  <h2>
    My To Do List
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
    <input
      type="submit"
      value="Add ToDo"
    />
  </form>
</div>
`;
